### ISSUE TITLE
Poetry export does not export local editable requirements correctly

### ISSUE THREAD

**Issue by pawelrubin** [python-poetry/poetry#3468]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS Catalina 10.15.7
- **Poetry version**: 1.1.4
- **Python version**: 3.9.0
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/pawelrubin/c20a783832924952dc0c0b9fa88974f0
- **Link of a [Gist](https://gist.github.com/) with the contents of your poetry.lock file**: 
https://gist.github.com/pawelrubin/19731324b4ed3b1f3d50c6634befc016
- **Link of a [Gist](https://gist.github.com/) with the contents of your example_pkg/setup.py file**: 
https://gist.github.com/pawelrubin/563bba1d591959c674c8aca81d3918ad

## Issue

Local editable packages are not exported correctly by poetry. Thus, `pip install -r requirements.txt` fails with error:
```shell
ERROR: Invalid requirement: 'example-pkg @ /Users/pawelvewd/projects/test_poetry/example_pkg' (from line 3 of requirements.txt)
Hint: It looks like a path. File 'example-pkg @ /Users/pawelvewd/projects/test_poetry/example_pkg' does not exist.
```

### Expected result
```shell
-e ./example_pkg["foo"] ; python_version >= "3.9"
```

### Current result
```shell
example-pkg @ /Users/pawelvewd/projects/test_poetry/example_pkg; python_version >= "3.9"
```

- the path should be relative IMO
- Poetry does not export `extras`
- Poetry does not export `develop = true` (editable mode)

**Comment by gitpushdashf** [comment]
I ran into this as well. Thanks for posting the bug.

**Comment by ShayNehmad-RecoLabs** [comment]
I need this as well! Please let us know what's up with this issue.

**Comment by lee-hodg** [comment]
Yep, I was just about to create an issue for this. Thanks for posting.

**Comment by stygmate** [comment]
same here sticking with 1.0.0 version

**Comment by bow** [comment]
I just stumbled upon this as well (Poetry 1.1.4, Python 3.9.1). In my case, I made the generated `requirements.txt` work well with pip by changing the path into a [direct reference](https://www.python.org/dev/peps/pep-0440/#direct-references) (My use case was not strictly only local development)

Would it make sense to have this behavior for path dependencies?

* If `develop` = true, create a `-e <path>` line (as noted in the issue description)
* If `develop` = false, create a direct reference

**Comment by justinvanwinkle** [comment]
is poetry still being actively developed?  Lack of support for relative paths is a huge barrier to adoption.

**Comment by fkaleo** [comment]
> I just stumbled upon this as well (Poetry 1.1.4, Python 3.9.1). In my case, I made the generated `requirements.txt` work well with pip by changing the path into a [direct reference](https://www.python.org/dev/peps/pep-0440/#direct-references) (My use case was not strictly only local development)
> 
> Would it make sense to have this behavior for path dependencies?
> 
>     * If `develop` = true, create a `-e <path>` line (as noted in the issue description)
> 
>     * If `develop` = false, create a direct reference

Thanks for the idea @bow ! Would you have an example of how exactly you did it, ie. what you wrote in your `pyproject.toml`?

**Comment by bow** [comment]
@fkaleo It was actually done outside of Poetry ~ simply using `sed` to add a `file://` prefix to the generated path.

I should say first, I just tested this below again in Poetry 1.1.7 and I could see that it already creates direct references for the local dependency. So it seems like the trick below is not necessary anymore. My use case was also to generate wheels from `requirements.txt` (i.e. `pip wheel`) and not `pip install`.

That being said, with something like this in my `pyproject.toml` (`local-pkg has a regular `pyproject.toml`):
```
...
[tool.poetry.dependencies]
local-pkg = { path = "/path/to/local/pkg" }
```
it was simply this
```
poetry export -f requirements.txt -o /tmp/requirements.txt
  && sed -i 's/local-pkg @ /local-pkg @ file:\/\//1' /tmp/requirements.txt
  && pip wheel -r /tmp/requirements.txt --wheel-dir=/path/to/a/dir
```
